src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -197,14 +197,26 @@ protected override void Process ()
 			ProcessXml (Context.StripLinkAttributes, Context.IgnoreLinkAttributes);
 		}
 
-		protected override bool AllowAllAssembliesSelector { get => true; }
+		protected override AllowedAssemblies AllowedAssemblySelector {
+			get {
+				if (_resourceAssembly == null)
+					return AllowedAssemblies.AllAssemblies;
 
-		protected override void ProcessAssembly (AssemblyDefinition assembly, XPathNodeIterator iterator, bool warnOnUnresolvedTypes)
+				// Corelib XML may contain assembly wildcard to support compiler-injected attribute types
+				string coreLib = _resourceAssembly.MainModule.TypeSystem.CoreLibrary.Name;
```

Comment:
We should hardcode this to `System.Private.CoreLib` only, no point to allow things like netstandard.dll